cmd/snap-confine/snap-confine.c

```diff
@@ -81,6 +81,9 @@ int main(int argc, char **argv)
 	if (!verify_security_tag(security_tag))
 		die("security tag %s not allowed", security_tag);
 
+	const char *snap_name = getenv("SNAP_NAME");
+	sc_snap_name_validate(snap_name, NULL);
```

Comment:
It feels very magic (to me) that `sc_snap_name_validate(snap_name, NULL)` will die() on the wrong names. After reading the code I see that its because of the passed `NULL` but that does not feel intuitive to me.